### PR TITLE
[BugFix] Fix uncorrect index id for GIN index file

### PR DESCRIPTION
**Pull request by srlch**
## Why I'm doing:
Index id in TabletIndex and offset of a index in tablet schema maybe different. Because the Bitmap Index index id in TabletIndex is always -1(by design). In rowset::link_files_to we use offset instead of index_id() to generated gin index file path which is wrong.

## What I'm doing:
Make sure use index id in TabletIndex to get the index file path.

Fixes #issue

## What type of PR is this:

- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.3
  - [ ] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

**Review by wyb — DISMISSED** [review]

File: be/src/storage/rowset/rowset.cpp [hunk 1]
```diff
@@ -400,9 +400,9 @@ Status Rowset::link_files_to(KVStore* kvstore, const std::string& dir, RowsetId
                 const auto& index = (*(_schema->indexes()))[index_id];
```
Comment (trueeyu) [inline be/src/storage/rowset/rowset.cpp:400]: Rename index_id to index?

**Review by trueeyu — DISMISSED** [review]

File: be/src/storage/rowset/rowset.cpp (hunk 1 above)
Comment (srlch) [reply be/src/storage/rowset/rowset.cpp]: Fix it, use i instead

File: be/src/storage/rowset/rowset.cpp [hunk 2]
```diff
@@ -396,13 +396,13 @@ Status Rowset::link_files_to(KVStore* kvstore, const std::string& dir, RowsetId
         // link inverted files
         if (!_schema->indexes()->empty()) {
             int segment_n = i;
-            for (int index_id = 0; index_id < _schema->indexes()->size(); index_id++) {
-                const auto& index = (*(_schema->indexes()))[index_id];
+            for (int i = 0; i < _schema->indexes()->size(); i++) {
```
Comment (wyb) [inline be/src/storage/rowset/rowset.cpp:399]: `i` is already used

Comment (srlch) [reply be/src/storage/rowset/rowset.cpp]: Fixed

**Review by wyb — APPROVED** [review]

**Review by trueeyu — APPROVED** [review]

**Pull request merged**